commands/http/parse.go

```diff
@@ -20,12 +20,20 @@ func Parse(r *http.Request, root *cmds.Command) (cmds.Request, error) {
 
 	stringArgs := make([]string, 0)
 
+	if err := apiVersionMatches(r); err != nil {
```

Comment:
we could define an API version in the root or something... not sure how this does with command trees and so on. seems like a bigger change than we want this PR to be. (if we want it for this one, then do that PR first and rebase this on that one... but i would say just do the simpler thing now)